### PR TITLE
Add my subscription page

### PR DESCRIPTION
**Pull request by barshathakuri**
## Addresses:
- #150 
- #151 

## Depends on:
- XXX

## Changes
- Add subscription page
- Add subscribe tab
- Add archive tab

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [ ] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [ ] codegen errors

### CODE REVIEW

File: src/App/routes/index.tsx [hunk 1]
```diff
@@ -57,6 +57,20 @@ const homeLayout = customWrapRoute({
...
 
+const mySubscription = customWrapRoute({
+    parent: rootLayout,
+    path: 'subscription',
+    component: {
+        render: () => import('#views/MySubscription'),
+        props: {},
+    },
+    context: {
+        title: 'My Subscription',
```
Comment (AdityaKhatri) [inline src/App/routes/index.tsx]:
```suggestion
        title: 'My Subscriptions',
```

File: src/App/routes/index.tsx [hunk 2]
```diff
@@ -57,6 +57,20 @@ const homeLayout = customWrapRoute({
     },
 });
 
+const mySubscription = customWrapRoute({
+    parent: rootLayout,
+    path: 'subscription',
```
Comment (AdityaKhatri) [inline src/App/routes/index.tsx]:
```suggestion
    path: 'subscriptions',
```

File: src/components/Navbar/i18n.json [hunk 3]
```diff
@@ -6,6 +6,8 @@
         "appAbout": "About",
         "appResources": "Resources",
         "headerMenuHome": "Home",
-        "headerMenuMySubscription": "My Subscription"
+        "headerMenuMySubscription": "My Subscription",
```
Comment (AdityaKhatri) [inline src/components/Navbar/i18n.json]:
```suggestion
        "headerMenuMySubscription": "My Subscriptions",
```

File: src/views/Home/AlertsMap/index.tsx [hunk 4]
```diff
@@ -77,6 +83,11 @@ type AlertPointProperties = {
 export function Component() {
     const strings = useTranslation(i18n);
     const alertFilters = useAlertFilters();
+    const [isModalOpen, setModalOpen] = useState(false);
+
+    const openModal = () => setModalOpen(true);
+    const closeModal = () => setModalOpen(false);
```
Comment (AdityaKhatri) [inline src/views/Home/AlertsMap/index.tsx]: use useModalState or useBooleanState

File: src/views/Home/AlertsTable/index.tsx [hunk 5]
```diff
@@ -104,6 +110,11 @@ const DESC = 'DESC';
 export function Component() {
     const strings = useTranslation(i18n);
 
+    const [isModalOpen, setModalOpen] = useState(false);
+
+    const openModal = () => setModalOpen(true);
+    const closeModal = () => setModalOpen(false);
```
Comment (AdityaKhatri) [inline src/views/Home/AlertsTable/index.tsx]: same as before

File: src/views/MySubscription/ActiveSubscription/SubscriptionDetail/SubscriptionActions/index.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import { MoreOptionsIcon } from '@ifrc-go/icons';
```
Comment (AdityaKhatri) [inline src/views/MySubscription/ActiveSubscription/SubscriptionDetail/SubscriptionActions/index.tsx]: Unnecessary separation of component

File: src/views/MySubscription/ActiveSubscription/SubscriptionDetail/index.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import { Container } from '@ifrc-go/ui';
+
+import SubscriptionActions from './SubscriptionActions';
+
+import styles from './styles.module.css';
+
+interface Props {
+    data: any;
```
Comment (AdityaKhatri) [inline src/views/MySubscription/ActiveSubscription/SubscriptionDetail/index.tsx]: define data structure of Subscription

File: src/views/MySubscription/ActiveSubscription/index.tsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+import styles from './styles.module.css';
+
+function ActiveTable() {
+    const [data, setData] = useState([
+        {
+            id: 1,
+            alertName: 'Flood',
+            filtersApplied: [
+                'North America',
```
Comment (AdityaKhatri) [inline src/views/MySubscription/ActiveSubscription/index.tsx]: define Subscription data type. Only then create dummy data for that.

File: src/views/MySubscription/ActiveSubscription/index.tsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+                'Asia-Pacific',
+                'Nepal',
+                'Critical',
+            ],
+            totalAlertCount: 200,
+        },
+    ]);
+
+    const handleDelete = useCallback((id: any) => {
+        setData((prevData) => prevData.filter((item) => item.id !== id));
```
Comment (AdityaKhatri) [inline src/views/MySubscription/ActiveSubscription/index.tsx]: not necessary

File: src/views/MySubscription/ActiveSubscription/index.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    }, []);
+
+    const handleArchive = useCallback((id: any) => {
+        setData((prevActiveData) => {
+            const itemToArchive = prevActiveData.find((item) => item.id === id);
+            if (isNotDefined(itemToArchive)) return prevActiveData;
+            setData((prevArchivedData) => [...prevArchivedData, itemToArchive]);
+            return prevActiveData.filter((item) => item.id !== id);
+        });
```
Comment (AdityaKhatri) [inline src/views/MySubscription/ActiveSubscription/index.tsx]: not necessary

File: src/views/MySubscription/ActiveSubscription/index.tsx [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    }, []);
+
+    const rendererParams = useCallback((_: string, value: any) => ({
+        data: value,
+        onDelete: () => handleDelete(value.id),
+        onArchive: () => handleArchive(value.id),
```
Comment (AdityaKhatri) [inline src/views/MySubscription/ActiveSubscription/index.tsx]: set it directly, send id from inside the component

File: src/views/MySubscription/ActiveSubscription/index.tsx [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+        <Container
+            className={styles.activeSubscription}
+            contentViewType="vertical"
+        >
+            <List
+                className={styles.subscription}
+                data={data}
+                renderer={SubscriptionDetail}
+                rendererParams={rendererParams}
+                keySelector={(data) => data.id}
```
Comment (AdityaKhatri) [inline src/views/MySubscription/ActiveSubscription/index.tsx]: never do this

File: src/views/MySubscription/ArchiveSubscription/SubscriptionDetail/SubscriptionActions/index.tsx [hunk 13]
```diff
@@ -0,0 +1,43 @@
+import { MoreOptionsIcon } from '@ifrc-go/icons';
```
Comment (AdityaKhatri) [inline src/views/MySubscription/ArchiveSubscription/SubscriptionDetail/SubscriptionActions/index.tsx]: unnecessary components

File: src/views/MySubscription/ArchiveSubscription/SubscriptionDetail/index.tsx [hunk 14]
```diff
@@ -0,0 +1,38 @@
+import { Container } from '@ifrc-go/ui';
```
Comment (AdityaKhatri) [inline src/views/MySubscription/ArchiveSubscription/SubscriptionDetail/index.tsx]: create a single one and reuse in both

File: src/views/MySubscription/ArchiveSubscription/SubscriptionDetail/index.tsx [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+
+import SubscriptionActions from './SubscriptionActions';
+
+import styles from './styles.module.css';
+
+interface Props {
+    data: any;
```
Comment (AdityaKhatri) [inline src/views/MySubscription/ArchiveSubscription/SubscriptionDetail/index.tsx]: send necessary data individually

File: src/views/MySubscription/index.tsx [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+export function Component() {
+    const strings = useTranslation(i18n);
+
+    type TabKey = 'active' | 'archive';
+    const [activeTab, setActiveTab] = useState<TabKey>('active');
+    const [isModalOpen, setModalOpen] = useState(false);
+
+    const openModal = () => setModalOpen(true);
+
+    const closeModal = () => setModalOpen(false);
```
Comment (AdityaKhatri) [inline src/views/MySubscription/index.tsx]: do not do this

File: src/views/NewSubscriptionModal/index.tsx [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+    const strings = useTranslation(i18n);
+
+    const [isCheckboxChecked, setCheckboxChecked] = useState(false);
+    const [selectedValue, setSelectedValue] = useState('daily');
+    const [alertTitle, setAlertTitle] = useState('');
+
+    const handleCheckboxChange = (checked: any) => {
+        setCheckboxChecked(checked);
+    };
+
```
Comment (AdityaKhatri) [inline src/views/NewSubscriptionModal/index.tsx]: create a proper form based on the created subscription data structure

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: src/App/routes/index.tsx [hunk 18]
```diff
@@ -74,6 +88,20 @@ const homeIndex = customWrapRoute({
...
+        render: () => import('#views/Home'),
+        props: {},
+    },
+    context: {
+        title: 'Historical Alerts',
+        // TODO: Change visibility after login feature
+        visibility: 'anything',
+    },
+});
+
```
Comment (AdityaKhatri) [inline src/App/routes/index.tsx]: remove this

File: src/views/Home/AlertFilters/index.tsx [hunk 19]
```diff
@@ -118,7 +118,7 @@ query AllCountryList {
 `;
 
 interface Props {
-    variant: 'map' | 'table';
+    variant?: 'map' | 'table';
```
Comment (AdityaKhatri) [inline src/views/Home/AlertFilters/index.tsx]: undo

File: src/views/Home/AlertsMap/index.tsx [hunk 20]
```diff
@@ -77,6 +85,12 @@ type AlertPointProperties = {
 export function Component() {
     const strings = useTranslation(i18n);
     const alertFilters = useAlertFilters();
+
+    const [showExportModal, {
+        setTrue: setShowExportModalTrue,
+        setFalse: setShowExportModalFalse,
+    }] = useBooleanState(false);
```
Comment (AdityaKhatri) [inline src/views/Home/AlertsMap/index.tsx]: rename

File: src/views/Home/AlertsTable/index.tsx [hunk 21]
```diff
@@ -104,6 +112,11 @@ const DESC = 'DESC';
 export function Component() {
     const strings = useTranslation(i18n);
 
+    const [showExportModal, {
+        setTrue: setShowExportModalTrue,
+        setFalse: setShowExportModalFalse,
+    }] = useBooleanState(false);
```
Comment (AdityaKhatri) [inline src/views/Home/AlertsTable/index.tsx]: rename

File: src/views/NewSubscriptionModal/index.tsx [hunk 22]
```diff
@@ -0,0 +1,396 @@
...
+const ALL_COUNTRY_LIST = gql`
+query AllCountryList {
+  public {
+    id
+    allCountries {
+      name
+      id
+    }
+  }
```
Comment (AdityaKhatri) [inline src/views/NewSubscriptionModal/index.tsx]: all enums here as well, fix indent

File: src/views/NewSubscriptionModal/index.tsx [hunk 23]
```diff
@@ -0,0 +1,396 @@
...
+    key: string,
+    label: string,
+};
+
+interface AlertFilters {
+    key: string;
+    label: string;
+}
+
+interface SubscriptionForm {
```
Comment (AdityaKhatri) [inline src/views/NewSubscriptionModal/index.tsx]: add TODO to use this from generated after API is complete

File: src/views/NewSubscriptionModal/index.tsx [hunk 24]
```diff
@@ -0,0 +1,396 @@
...
+    id: string;
+    title: string;
+    urgency: AlertInfoUrgencyEnum[] | undefined;
+    severity: AlertInfoSeverityEnum[] | undefined;
+    certainty: AlertInfoCertaintyEnum[] | undefined;
+    country: string | undefined;
+    admin1: string | undefined;
+    sendEmail: boolean;
+    frequency: string | undefined;
```
Comment (AdityaKhatri) [inline src/views/NewSubscriptionModal/index.tsx]:
```suggestion
    frequency: 'daily' | 'weekly' | undefined;
```

File: src/views/NewSubscriptionModal/index.tsx [hunk 25]
```diff
@@ -0,0 +1,396 @@
...
+    const defaultFormValue: PartialFormFields = {
+        title: subscription?.title,
+        urgency: subscription?.urgency,
+        severity: subscription?.severity,
+        certainty: subscription?.certainty,
+        sendEmail: subscription?.sendEmail,
+        frequency: subscription?.frequency,
+        country: subscription?.country,
+        admin1: subscription?.admin1,
+    };
```
Comment (AdityaKhatri) [inline src/views/NewSubscriptionModal/index.tsx]: put this under useMemo

File: src/views/NewSubscriptionModal/index.tsx [hunk 26]
```diff
@@ -0,0 +1,396 @@
...
+        activeCountryId,
+        selectedSeverityTypes,
+        selectedUrgencyTypes,
+        selectedCertaintyTypes,
+        setActiveCountryId,
+        setSelectedSeverityTypes,
+        setSelectedUrgencyTypes,
+        setSelectedCertaintyTypes,
+        // setStartDateFrom,
+        // setStartDateTo,
```
Comment (AdityaKhatri) [inline src/views/NewSubscriptionModal/index.tsx]: remove

File: src/views/NewSubscriptionModal/index.tsx [hunk 27]
```diff
@@ -0,0 +1,396 @@
...
+        data: alertEnumsResponse,
+    } = useQuery<AlertEnumsQuery, AlertEnumsQueryVariables>(
+        ALERT_ENUMS,
+    );
+
+    const {
+        data: allCountryListResponse,
+    } = useQuery<AllCountryListQuery, AllCountryListQueryVariables>(
+        ALL_COUNTRY_LIST,
+    );
```
Comment (AdityaKhatri) [inline src/views/NewSubscriptionModal/index.tsx]: merge these two

File: src/views/NewSubscriptionModal/index.tsx [hunk 28]
```diff
@@ -0,0 +1,396 @@
...
+
+    const {
+        data: adminResponse,
+    } = useQuery<FilteredAdminListQuery, FilteredAdminListQueryVariables>(
+        ADMIN_LIST,
+        { variables: adminQueryVariables, skip: isNotDefined(value.country) },
+    );
+
+    const subscriptionCreate = () => {
```
Comment (AdityaKhatri) [inline src/views/NewSubscriptionModal/index.tsx]: use useCallback

File: src/views/NewSubscriptionModal/index.tsx (hunk 26 above)
Comment (AdityaKhatri) [inline src/views/NewSubscriptionModal/index.tsx]: do not update this, completely remove

File: src/views/MySubscription/index.tsx [hunk 29]
```diff
@@ -0,0 +1,160 @@
...
+        },
+    ];
+
+    type TabKey = 'active' | 'archive';
+    const [activeTab, setActiveTab] = useState<TabKey>('active');
+
+    const [showExportModal, {
+        setTrue: setShowExportModalTrue,
+        setFalse: setShowExportModalFalse,
+    }] = useBooleanState(false);
```
Comment (AdityaKhatri) [inline src/views/MySubscription/index.tsx]: rename

File: src/views/MySubscription/index.tsx [hunk 30]
```diff
@@ -0,0 +1,160 @@
...
+                        )}
+                    >
+                        {strings.myNewSubscription}
+                    </Button>
+                )}
+            >
+                {showExportModal && (
+                    <NewSubscriptionModal
+                        onCloseModal={setShowExportModalFalse}
+                        subscription={undefined}
```
Comment (AdityaKhatri) [inline src/views/MySubscription/index.tsx]: send current subscription

File: src/views/MySubscription/index.tsx [hunk 31]
```diff
@@ -0,0 +1,160 @@
...
+
+    type TabKey = 'active' | 'archive';
+    const [activeTab, setActiveTab] = useState<TabKey>('active');
+
+    const [showExportModal, {
+        setTrue: setShowExportModalTrue,
+        setFalse: setShowExportModalFalse,
+    }] = useBooleanState(false);
+
+    const rendererParams = useCallback((_: string, value: SubscriptionDetail) => ({
```
Comment (AdityaKhatri) [inline src/views/MySubscription/index.tsx]: use same renderer but different rendererParams

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: src/views/NewSubscriptionModal/index.tsx (hunk 22 above)
Comment (barshathakuri) [reply src/views/NewSubscriptionModal/index.tsx:95]: There is no AllCountryList in enums.

File: src/views/MySubscription/SubscriptionTableItem/index.tsx [hunk 32]
```diff
@@ -0,0 +1,60 @@
...
+                    {actions}
+                </div>
+            </div>
+            <div className={styles.filterApplied}>
+                {country}
+                {admin1}
+                {urgency}
+                {certainty}
+                {severity}
+            </div>
```
Comment (AdityaKhatri) [inline src/views/MySubscription/SubscriptionTableItem/index.tsx]: put this inside container

File: src/views/NewSubscriptionModal/index.tsx [hunk 33]
```diff
@@ -0,0 +1,359 @@
...
+
+function NewSubscriptionModal(props: Props) {
+    const {
+        subscription,
+        onCloseModal,
+        urgency,
+        severity,
+        certainty,
+        activeCountry,
+        activeAdmin1,
```
Comment (AdityaKhatri) [inline src/views/NewSubscriptionModal/index.tsx]: this is either or

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**